0003-lightning-as-alpha-ledger.adoc

```diff
@@ -0,0 +1,279 @@
+However, timeouts on the HTLCs are relative and will get accumulated across the payment path, which makes it harder to predict.
+Hence, Alice will need to know the path in advance before sending a swap request to Bob as she won't (or should not) be able to change neither `alpha_expiry` nor `beta_expiry` after the swap request.
+
+Timeouts on the HTLCs are configured globally for the payee node and can optionally be defined when paying by the payer node (`--final_cltv_delta=T`).
+
+
+IMPORTANT: TODO: we should investigate this topic more in detail.
+
+=== Responsibilities
+
+A main goal of COMIT is to keep the autonomy to the user and let him/her decide when to deploy a HTLC, redeem or refund a HTLC, etc.
+If a trade involves LN using LND we can approach these things differently:
+
+
+* Action
+    ** Create hold invoice
+* Responsibility
+    ** LND
+* Invoker
+    ** User, comit-i, comit-node
+* Description
+    ** `addholdinvoice` is available as a RPC command or through the LND CLI. Although dealing with this is rather cumbersome, to keep the autonomy with the user, and to not introduce LND dependency into the comit-node, we this should be possible through comit-i.
+* Conclusion:
+    ** comit-i needs LND support. However, in order to do this, we will need to introduce a new action which is meant to be executed prior accepting a swap request:
+    *** Bob receives a swap request from Alice (an learns about the hashed secret)
+    *** *Action 1:* Bob creates a hold invoice through comit-i
+    *** *Action 2:* Bob accepts the swap requests by posting the newly generated invoice ID back to comit-rs
+    A quick research showed that LN payments can be done with the browser extension: https://lightningjoule.com/[Joule] and requests to a LND node can be done through the browser.
+
+---
+
+
+* Action
+    ** Pay invoice
+* Responsibility:
+    ** LND or LN Wallet
+* Invoker
+    ** User, comit-i
+* Description
+    ** To keep the autonomy to the user when to initiate a trade, we should return the invoice information through our API to the user (e.g. expose it through comit-i ) and let him/her pay the invoice.
+* Conclusion
+    ** comit-i needs LND support.
+
+
+
+---
+
+
+* Action:
+    ** Settle Invoice
+* Responsibility
+    ** LND or LN Wallet
+* Invoker
+    ** User, comit-i, comit-node
+* Description
+    ** As soon as the secret has been learned, the HTLC on the LN should be settled using the command `settleinvoice`, this can either be done by the user (and exposed through comit-i) or done automatically through the comit-node. Since we have the extra _redeem_ step for Bitcoin and Ethereum (as well for Erc20) which needs to be performed from the user, we should leave the settlement of the invoice to the user (e.g. expose this information through comit-i).
```

Comment:
An important thing to note here is that one of the reasons we have an extra redeem step is because we cannot send transactions from the comit-node (we don't have funds). For LN, this restriction is not given since we don't have to pay any fees to settle an invoice.

I am not suggesting that we should change our behaviour because of that. It is just something to keep in mind. Once we have L1 support, it could be worth considering to give the comit-node the functionality to own little funds, enough for sending redeem transactions. For L2 ledgers, we could then also automatically redeem funds.

Reply:
Another point is that not doing the automated settlement allows Alice to use her option whereas if we automate it, Alice would have to use a different implementation to use her option (or fork comit-rs)